core/src/main/java/org/elasticsearch/index/mapper/internal/TypeFieldMapper.java

```diff
@@ -105,6 +126,29 @@ public String typeName() {
             return CONTENT_TYPE;
         }
 
+        public boolean fielddata() {
+            return fielddata;
+        }
+
+        public void setFielddata(boolean fielddata) {
```

Comment:
can you add a test for it in TypeFieldTypeTests.setupProperties? (see for instance how DateFieldTypeTests does it)